robot_log_visualizer/robot_visualizer/meshcat_provider.py

```diff
@@ -138,34 +139,71 @@ def find_model_joints(model_name, considered_joints):
 
                     if model_filenames:
                         model_found_in_env_folders = True
-                        self.custom_model_path = str(model_filenames[0])
+                        self.model_path = str(model_filenames[0])
                         break
 
             # If the model is not found we exit
             if not model_found_in_env_folders:
                 return False
 
             self.model_joints_index = find_model_joints(
-                self.custom_model_path, considered_joints
+                self.model_path, considered_joints
             )
             considered_model_joints = [
                 considered_joints[i] for i in self.model_joints_index
             ]
             model_loader.loadReducedModelFromFile(
-                self.custom_model_path, considered_model_joints
+                self.model_path, considered_model_joints
             )
 
         if not model_loader.isValid():
             return False
 
+        self.meshcat_visualizer_mutex.lock()
+
+        # if the model is already loaded we remove it
+        if self._is_model_loaded:
+            self._meshcat_visualizer.delete(shape_name="robot")
+            self._is_model_loaded = False
+
+        model = model_loader.model()
+        if base_frame is None:
+            self.frame_T_base = np.eye(4)
+            self.base_frame = model.getFrameName(model.getDefaultBaseLink())
+            link_frame = None
+        else:
+            base_frame_index = model.getFrameIndex(base_frame)
```

Comment:
The index can be invalid if the link is not found in the model

Reply:
I can raise an exception here. However, notice the cause you mention will not ever happen since the GUI provides the base frame and will be one of the framea of the robot taken from

https://github.com/ami-iit/robot-log-visualizer/blob/9c40bb514067c039b5c762420a066073f6ef118c/robot_log_visualizer/robot_visualizer/meshcat_provider.py#L193

Reply:
Done in https://github.com/ami-iit/robot-log-visualizer/pull/93/commits/17df7943a246a626d82bbb3d40e46c272795d6b8

Reply:
> I can raise an exception here. However, notice the cause you mention will not ever happen since the GUI provides the base frame and will be one of the framea of the robot taken from
> 
> https://github.com/ami-iit/robot-log-visualizer/blob/9c40bb514067c039b5c762420a066073f6ef118c/robot_log_visualizer/robot_visualizer/meshcat_provider.py#L193

Ah I did not notice it!